docs/DeveloperGuide.md

```diff
@@ -195,6 +195,65 @@ Therefore, given the constraints of the UI, sorting is implemented as a **perman
    * Pros: Sorted result is not permanent. So sorting won't interfere with the underlying persons list.
    * Cons: Requires redesigning a significant part of the `UI`, `Logic` and `Model` classes.
 
+### Adding Events
+
+The Add Event feature that is accessed through the `addEvent` command allows users to add new marketing campaigns of 
+the `event` class to the application. 
+
+The `event` added by the user will have 4 compulsory fields:
+- Title of the `event`
+- Date of the `event`
+- Time of the `event`
+- Purpose of the `event`
+
+The `addEvent` operation is facilitated by `AddEventCommand` which extends from `Command`. If the users' input matches
+the COMMAND_WORD of `AddEventCommand` in `AddressBookParser#parseCommand()`, `AddEventCommandParser#parse()` will 
+process the additional user inputs which constitute the 4 compulsory fields of the `event` class and return an 
+`AddEventCommand`. 
+
+Executing this Command object through the `AddEventCommand#execute()` triggers the `Model` interface's 
+`Model#addEvent()`. This operation subsequently calls upon the `AddressBook#addEvent()` operation which in turn call 
+upon the `UniqueEventList#add()` operation and the `event` will be stored in memory.
+
+The addEvent operation will also trigger the `StorageManager#saveAddressBook()` operation which will save the event to
+a json format together with all other `person`(s) and `event`(s) in memory.
+
+The following sequence diagram will illustrate how the `addEvent` operation works: 
+
+![AddEventSequenceDiagram](images/AddEventSequenceDiagram.png)
+
+<div markdown="span" class="alert alert-info">:information_source: **Note:** `cmd` in the diagram represents the add
+event command text entered by user. a represents the AddEventCommand instance and b represents the ModelManager instance
+while event represents the instance of event class created and stored within AddEventCommand. 
+Additionally, saving of the updated events list has been excluded from this diagram for simplicity.
+</div>
```

Comment:
Would it help make your DG simpler by removing this information_source since the UML diagram already explains what you are mentioning here?

Reply:
Thanks for all the feedback! Ok, have removed the middle sentence "a represents ... ModelManager instance" since that is immediately apparent from the diagram. However, I feel that leaving the other parts in may help the reader understand the actual input/output better since its not fully explained in the diagram.